docs/BacwardCompatibility.md

```diff
@@ -0,0 +1,133 @@
+### Relations between CRD changes and Fybrik releases
+Based on the above definitions, the relation between CRD versions and Fybrik versions is unidirectional. Which means,
+CRD API changes requires at least minor Fybrik release, on the other hand not each Fybrik release requires a new CRD
+version.
+
+### Upgrade Plan
+The CustomResourceDefinition API provides a workflow for introducing and upgrading to new versions of a 
+CustomResourceDefinition. For more information about 
+[serving multiple versions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#serving-multiple-versions-of-a-crd) of your CustomResourceDefinition 
+and migrating your objects from one version to another see 
+[Versions in CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/)
+
+The first step of adding a new version is to "pick a conversion strategy. Since custom resource objects need to be able 
+to be served at both versions, that means they will sometimes be served at a different version than their storage 
+version. In order for this to be possible, the custom resource objects must sometimes be converted between the version 
+they are stored at and the version they are served at. If the conversion involves schema changes and requires custom 
+logic, a conversion **Webhook** should be used. If there are no schema changes, the default **None** conversion strategy may 
+be used and only the apiVersion field will be modified when serving different versions."
+
+Due to some Fybrik customers' limitations, we cannot use Webhooks in Fybrik deployments. Therefore, below we investigate 
+possible CRD changes without changing its schema.
+
+We were able to change a required field in version v1 to an optional field in v2. In addition, we were able to add a new 
+optional field in v2. This is inline with compatibility examples from [\[5\]](#5)
+
+*Note:* If we go from previous API version to a new one without supporting the previous version, let say `v1alpha1` and 
+a cluster API server has stored objects of the previous version, we will get an error similar to the following:  
+```
+CustomResourceDefinition.apiextensions.k8s.io "plotters.app.fybrik.io" is invalid: status.storedVersions[0]: Invalid value: "v1alpha1": must appear in spec.versions
+```
+
+## Data Catalog and Policy Manager Connector Interfaces
+Fybrik manager communicates with connectors (data catalog and policy manager) over REST API. 
+
+REST doesn’t provide any specific versioning guidelines, however the more commonly used approaches fall into three 
+categories:
+- URI Versioning
+- Versioning using Custom Request Header
+- Versioning using the “Accept” header
+  For examples and explanations, see [REST API Versioning](https://restfulapi.net/versioning/)
+Traditionally, REST APIs do not the support semantic version model, each new version is a single number. 
```

Comment:
If I understand correctly, it's not that they don't support semantic versioning but rather that the REST API version is only meant to change when the backward INCOMPATIBLE changes - i.e. changes to major versions.

Reply:
Looks like you're right